api/tests.py

```diff
@@ -77,80 +77,88 @@ def test_get_matches_basic(self):
         self.assertCountEqual(
-            res.json(),
-            [
-                {"username": "cad"},
-                {"username": "dig"},
-                {"username": "fog"},
-                {"username": "gil"},
-            ],
+            usernames(res.json()),
+            ("cad", "dig", "fog", "gil"),
         )
         self.client.login(username="fog", password="password")
-        res = self.client.get("/api/match/", format="json")
+        res = self.client.get(endpoint, format="json")
         self.assertEqual(res.status_code, 200)
         self.assertCountEqual(
-            res.json(),
-            [
-                {"username": "boy"},
-                {"username": "cad"},
-                {"username": "dig"},
-                {"username": "eel"},
-                {"username": "gil"},
-            ],
+            usernames(res.json()),
+            ("boy", "cad", "dig", "eel", "gil"),
         )
         self.client.login(username="gil", password="password")
-        res = self.client.get("/api/match/", format="json")
+        res = self.client.get(endpoint, format="json")
         self.assertEqual(res.status_code, 200)
         self.assertCountEqual(
-            res.json(),
-            [
-                {"username": "ale"},
-                {"username": "boy"},
-                {"username": "cad"},
-                {"username": "dig"},
-                {"username": "eel"},
-                {"username": "fog"},
-            ],
+            usernames(res.json()),
+            ("ale", "boy", "cad", "dig", "eel", "fog"),
+        )
+
+    def test_update_match_status(self):
+        """
+        Check that a match status can be updated.
+        """
+        self.client.login(username="ale", password="password")
+        res = self.client.put(
+            "/api/matches/", {"matchId": 2, "response": False}, format="json"
         )
+        self.assertEqual(res.status_code, 200)
+        obj = MatchStatus.objects.get(user_lo_id=1, user_hi_id=2)
+        self.assertEqual(obj.user_lo_response, False)
+        res = self.client.put(
+            "/api/matches/", {"matchId": 2, "response": True}, format="json"
+        )
+        self.assertEqual(res.status_code, 200)
+        obj = MatchStatus.objects.get(user_lo_id=1, user_hi_id=2)
+        self.assertEqual(obj.user_lo_response, True)
+
+        self.client.login(username="boy", password="password")
+        res = self.client.put(
+            "/api/matches/", {"matchId": 1, "response": False}, format="json"
+        )
+        self.assertEqual(res.status_code, 200)
+        obj = MatchStatus.objects.get(user_lo_id=1, user_hi_id=2)
+        self.assertEqual(obj.user_hi_response, False)
+        res = self.client.put(
+            "/api/matches/", {"matchId": 1, "response": True}, format="json"
+        )
+        self.assertEqual(res.status_code, 200)
+        obj = MatchStatus.objects.get(user_lo_id=1, user_hi_id=2)
+        self.assertEqual(obj.user_hi_response, True)
```

Comment:
I'm just clarifying for myself (because I'm learning a lot of this) what is happening in this test. The test is to see how the system responds when one or both users respond to the match?

Reply:
Yea its just testing that the endpoint will update the match status table to reflect the new state of the match between two users.

Reply:
Okay, does it check what happens if only one person responds? Sorry, not great yet at reading the code.

Reply:
It checks that a user_lo and user_hi can both accept and reject a match.